### PR TITLE
Add Entry: C++ Maps 

### PR DESCRIPTION
**Pull request by KTom101**
New entry on C++ Maps and terms; keyword 19.

### Type of Change

- Adding a new entry
- Editing an existing entry (fixing a typo, bug, issues, etc)
- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:121]:
```suggestion
std::map <type1, type2,  std::greater<dataType>> mapName { {key1, value1}, {key2, value2}, ...};
```

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:23]:
```suggestion
std::map <type1, type2> mapName;
```

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:31]:
```suggestion
std::map <type1, type2> mapName { {key1, value1}, {key2, value2}, ...};
```

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:47]:
```suggestion
  std::map <std::string, int> emptyMap;
```

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:50]:
```suggestion
  std::map <std::string, int> clothingStore {{"tshirt", 10}, {"pants", 12}, {"sweaters", 18}};
```

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:69]:
```suggestion
std::cout << clothingStore["sweaters"]; // Output: 18
```

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:89]:
```suggestion
  std::map <int, std::string> :: iterator iter;
```

File: content/cpp/concepts/maps/terms/clear/clear.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/clear/clear.md:36]:
```suggestion
  std::map <std::string, int> supplies {{"pen", 5}, {"pencils", 3}, {"paper", 12}};
```

File: content/cpp/concepts/maps/terms/clear/clear.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/clear/clear.md:42]:
```suggestion
  std::cout << supplies.size();
```

File: content/cpp/concepts/maps/terms/insert/insert.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/insert/insert.md:44]:
```suggestion
  std::map <std::string, int> :: iterator iter;
```

File: content/cpp/concepts/maps/terms/remove/remove.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/remove/remove.md:16]: Is it `.erase()` or `.remove()`?

File: content/cpp/concepts/maps/terms/remove/remove.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/remove/remove.md:37]:
```suggestion
  std::map <std::string, std::string> fields {{"zero", "0"}, {"one", "1"},  {"two", "2"}, {"three", "3"}};
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@KTom101, reviewed and left a few comments.

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:84]:
```suggestion
 std::map<int, std::string> reptiles {
```

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:115]:
```suggestion
std::map<type1, type2, std::greater<dataType>> mapName;
```

File: content/cpp/concepts/maps/maps.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/maps.md:138]:
```suggestion
 std::map<int, std::string, std::greater<int>> reptiles {
```

File: content/cpp/concepts/maps/terms/remove/remove.md
Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/remove/remove.md:2]:
```suggestion
Title: '.erase()'
```

Comment (SSwiniarski) [inline content/cpp/concepts/maps/terms/remove/remove.md:2]: If this is `.erase()` the file needs to be renamed.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]

**Comment by KTom101** [comment]
@SSwiniarski I believe I made all your changes including the one about the file name. Please make sure, though. I had an issue where some of the direct commits didn't go through. So I made some of the changes locally. Thanks!

**Review by SSwiniarski — APPROVED** [review]
Looks good for next review.

File: content/cpp/concepts/maps/terms/insert/insert.md
Comment (Dusch4593) [inline content/cpp/concepts/maps/terms/insert/insert.md:23]: Could we add a sentence or two here about the snippet above?

File: content/cpp/concepts/maps/terms/clear/clear.md
Comment (Dusch4593) [inline content/cpp/concepts/maps/terms/clear/clear.md:23]: Could we add a sentence or two here about the snippet above?

File: content/cpp/concepts/maps/maps.md
Comment (Dusch4593) [inline content/cpp/concepts/maps/maps.md:126]:
```suggestion
### Codebyte Example
```

File: content/cpp/concepts/maps/maps.md
Comment (Dusch4593) [inline content/cpp/concepts/maps/maps.md:124]:
```suggestion
The `dataType` for the comparison function above must be the same as `type1` which is the data type for the keys.
```

File: content/cpp/concepts/maps/maps.md
Comment (Dusch4593) [inline content/cpp/concepts/maps/maps.md:100]:
```suggestion
The snippet above outputs the following:
```

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@KTom101 Finished with my review and left some comments/suggestions.

**Comment by KTom101** [comment]
@Dusch4593 I think I've finished with your changes. Let me know if you want to switch things around.

**Review by Dusch4593 — APPROVED** [review]
@KTom101 LGTM! 👍🏻

**Comment by Dusch4593** [comment]
@yangc95 @caupolicandiaz does this PR need one more review before merge?

**Review by caupolicandiaz — COMMENTED** [review]
LGTM!

**Comment by Dusch4593** [comment]
Thank you so much for these new entries @KTom101! 🚀

Once the entries go live we can check them out with the links below:
C++: Maps - https://www.codecademy.com/resources/docs/cpp/maps
C++ Maps: .clear() - https://www.codecademy.com/resources/docs/cpp/maps/clear
C++ Maps: .erase() - https://www.codecademy.com/resources/docs/cpp/maps/erase
C++ Maps: .insert() - https://www.codecademy.com/resources/docs/cpp/maps/insert

**Pull request merged**
